src/cljam/io/util.clj

```diff
@@ -156,9 +157,53 @@
     #"(?i)\.fai$" :fai
     #"(?i)\.(fa|fasta|fas|fsa|seq|fna|faa|ffn|frn|mpfa)" :fasta
     #"(?i)\.2bit$" :2bit
+    #"(?i)\.tbi$" :tbi
     #"(?i)\.vcf" :vcf
     #"(?i)\.bcf$" :bcf
     #"(?i)\.bed" :bed
+    #"(?i)\.gff3?" :gff
     #"(?i)\.wig" :wig
     #"(?i)\.(bigWig|bw)" :bigwig
     (throw (IllegalArgumentException. "Invalid file type"))))
+
+(defn file-type-from-bytes
+  "Tries to detect a file format based on contents of the byte array `ba`.
+  The input byte array must be larger than 4 bytes. Note that detection of some
+  formats `#{:fasta :fastq :wig :fai :bed}` is based on naive heuristics and
+  thus can fail."
+  [^bytes ba]
+  (let [s (String. ba 0 (Math/min (int 64) (alength ba)))
+        i (-> (ByteBuffer/wrap ba)
+              (.order ByteOrder/LITTLE_ENDIAN)
+              (.getInt)
+              (bit-and 0xffffffff))]
```

Comment:
I didn't know this trick to coerce signed int to unsigned long 😲
But `Integer/toUnsignedLong` is a little more readable for this very purpose...?

Reply:
Thanks! Fixed in 2c2123f.